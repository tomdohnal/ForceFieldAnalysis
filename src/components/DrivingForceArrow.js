import React from 'react';
import PropTypes from 'prop-types';

import Arrow from './Arrow';

const DrivingForceArrow = ({ strength }) => (
  <Arrow
    strength={strength}
    color="#A5BB2B"
  />
);

DrivingForceArrow.propTypes = {
  strength: PropTypes.number.isRequired,
};

export default DrivingForceArrow;
