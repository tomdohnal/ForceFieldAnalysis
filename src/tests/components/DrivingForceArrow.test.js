import React from 'react';
import { shallow } from 'enzyme';

import DrivingForceArrow from '../../components/DrivingForceArrow';

describe('DrivingForceArrow', () => {
  it('renders correctly with force prop', () => {
    const tree = shallow(<DrivingForceArrow force={5} />);

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with force and color prop', () => {
    const tree = shallow(<DrivingForceArrow force={5} color="#FFF" />);

    expect(tree).toMatchSnapshot();
  });
});
