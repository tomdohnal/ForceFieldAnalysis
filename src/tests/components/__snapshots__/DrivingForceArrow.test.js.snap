// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrivingForceArrow renders correctly with force and color prop 1`] = `
<Connect(Arrow)
  color="#859900"
  force={5}
/>
`;

exports[`DrivingForceArrow renders correctly with force prop 1`] = `
<Connect(Arrow)
  color="#859900"
  force={5}
/>
`;
